lib/Dancer2/Plugin.pm

```diff
@@ -481,6 +481,10 @@ sub register_plugin {
                 my $app_dsl_cb = _find_consumer();
                 my $dsl        = $app_dsl_cb->();
 
+                {
+                    no strict 'refs';
+                    *{"${plugin_module}::dsl"} = sub {$dsl};
```

Comment:
We close over the DSL object fetched in line 482 above. There may be potential for a memory cycle.
(dsl => app => plugin => dsl arrangement).  I couldn't tickle that with the included tests in this Pr, but wondered if we should be defensive and weaken `$dsl` where its generated in line 482; viz
`Scalar::Util::weaken( my $dsl = $app_dsl_cb->() );`